hibernate-core/src/main/java/org/hibernate/Session.java

```diff
@@ -970,6 +970,26 @@ public interface Session extends SharedSessionContract, EntityManager {
 	 */
 	void clear();
 
+	/**
+	 * Return the persistent instances of the given entity class with the given identifiers
+	 * as a list. The position of an instance in the list matches the position of its identifier
+	 * in the given array, and the list contains a null value if there is no persistent instance
+	 * matching a given identifier. If an instance is already associated with the session, that
+	 * instance is returned. This method never returns an uninitialized instance.
+	 * <p>
+	 * Every object returned by {@code findAll()} is either an unproxied instance of the given
+	 * entity class, or a fully-fetched proxy object.
+	 *
+	 * @param entityType the entity type
+	 * @param ids the identifiers
+	 *
+	 * @return an ordered list of persistent instances, with null elements representing missing
+	 *         entities
+	 *
+	 * @since 6.5
+	 */
+	<E> List<E> findAll(Class<E> entityType, Object... ids);
```

Comment:
So just a convenience for `byMultipleIds(Class<?>).multiLoad(Object...)`? Maybe it's just me, but I don't think we should start "duplicating" API for convenience.

Reply:
I also usually agree, but the thing is that JPA 3.2 has gone in a quite different direction with `FindOption`s.

I initially proposed something more like our `XxxxLoadAccess` interfaces, but the `XxxOptions` objects were the direction Lukas wanted to take, and he talked me into it. Ultimately I think this was the right choice.

Therefore the concept here, [as described in the issue](https://hibernate.atlassian.net/browse/HHH-17948), is that in H7 we would allow `findAll()` to accept `FindOption`s. Yes, it's still "duplication" in some sense, but justifiable duplication, I would say.